### DISCUSSION TITLE
Can we remove a warning in visualizer?

### DISCUSSION THREAD

**Discussion by mzr1996** [open-mmlab/mmengine#919] [General]
The below warning is verbose, I just want to use the visualizer to draw some images, but don't want to save it anywhere. So I don't want to pass a `save_dir` argument.
https://github.com/open-mmlab/mmengine/blob/c712070c900c4320c8643727ee4ea0afebf17dd9/mmengine/visualization/visualizer.py#L165-L167

**Comment by mzr1996** [comment]
@Harold-lkk

**Comment by Harold-lkk** [comment]
Maybe setting the default of `save_dir` to "save_dir" instead of None?

**Reply by mzr1996** [reply]
> Maybe setting the default of `save_dir` to "save_dir" instead of None?

Since the `save_dir` is only used to make the path of every backend, I think you can raise this warning only if the `vis_backends` is not None.

**Comment by Luo-Yihang** [comment]
@mzr1996  I agree with you that this warning is unnecessary. It would be better to include this if branch only if the vis_backends is not None. I plan to raise a pull request to implement this change. Please let me know your thoughts on this.
